geoapps/inversion/components/preprocessing.py

```diff
@@ -0,0 +1,354 @@
+#  Copyright (c) 2023 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+
+from __future__ import annotations
+
+import uuid
+
+import numpy as np
+from geoh5py import Workspace
+from geoh5py.data import Data
+from geoh5py.objects import ObjectBase
+from geoh5py.shared.utils import is_uuid
+
+from geoapps.inversion.utils import calculate_2D_trend
+from geoapps.octree_creation.application import OctreeMesh
+from geoapps.shared_utils.utils import filter_xy, get_locations
+
+
+def window_data(
+    data_object: ObjectBase,
+    components: list[str],
+    data_dict: dict,
+    workspace: Workspace,
+    window_azimuth: float,
+    window_center_x: float,
+    window_center_y: float,
+    window_width: float,
+    window_height: float,
+    mesh: OctreeMesh,
```

Comment:
I think we can remove that feature. Too many potential side effects. The check to see if the mesh is rotated should be done by the driver instead.